element-connector/src/main/java/org/eclipse/californium/elements/CorrelationContextMatcher.java

```diff
@@ -39,4 +42,15 @@
 	 */
 	boolean isResponseRelatedToRequest(CorrelationContext requestContext, CorrelationContext responseContext);
 
+	/**
+	 * Check, if message should be sent out using the current correlation
+	 * context of the connector.
+	 * 
+	 * @param messageContext correlation context of message
+	 * @param connectorContext correlation context of connector
+	 * @return true, if message should be sent, false, if message should not be
+	 *         sent.
+	 */
+	boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext);
```

Comment:
Though it's extension point, the future will show, if it's used different.
One idea behind that, is the possiblity to offer a "backward compatible matcher", which only checks for received responses und ignores the correlation context on sending as it was until this PR.